File: python/45.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        if nums==None or len(nums)==0 or len(nums)==1:
            return 0
        step=0
        left=0
        right=0
        for i in range(len(nums)-1):
            right=max(right,i+nums[i])
            if right>len(nums)-2:
                break
            if i==left:
                step+=1
                left=right
        return step+1
                    
                    
                    

File: java/dp/45.java
class Solution {
    public int jump(int[] nums) {
        if (nums.length == 1)
            return 0;
        int m = nums[0];
        int rtn = 1;
        int m_next = m;
        for (int i = 0; i < nums.length; i++) {
            if (i <= m) {
                m_next = Math.max(m_next, nums[i] + i);

            } else {
                m = m_next;
                m_next = Math.max(m_next, nums[i] + i);
                rtn++;
            }
            // if (i <= m) {
            // m = Math.max(m, nums[i] + i);
            // }
            // if (m >= nums.length - 1) return ;
            if (m >= nums.length - 1)
                return rtn++;
        }
        return rtn++;

    }

}
